wandb/sdk/artifacts/artifact.py

```diff
@@ -2102,6 +2160,60 @@ def _load_manifest(self, url: str) -> None:
                 dep_artifact = entry._get_referenced_artifact(self._client)
                 self._dependent_artifacts.add(dep_artifact)
 
+    @staticmethod
+    def _get_gql_artifact_fragment() -> str:
+        fields = InternalApi().server_create_artifact_introspection()
+        fragment = """
+            fragment ArtifactFragment on Artifact {
+                id
+                artifactSequence {
+                    project {
+                        entityName
+                        name
+                    }
+                    name
+                }
+                versionIndex
+                artifactType {
+                    name
+                }
+                description
+                metadata
+                ttlDurationSeconds
+                aliases {
+                    artifactCollection {
+                        project {
+                            entityName
+                            name
+                        }
+                        name
+                    }
+                    alias
+                }
+                state
+                commitHash
+                fileCount
+                createdAt
+                updatedAt
+            }
+        """
+        if "ttlDurationSeconds" not in fields:
+            return fragment.replace("ttlDurationSeconds", "")
+        return fragment
+
+    @staticmethod
+    def _ttl_duration_seconds_to_gql(
+        ttl_duration_seconds: Optional[int], ttl_status: Optional[ArtifactTTLChange]
+    ) -> Optional[int]:
+        # Set artifact ttl value to ttl_duration_seconds if the user set a value
+        # otherwise use ttl_status to indicate to backend INHERIT or NOT_INHERIT the None ttl value
+        # Ttl_duration_seconds = None and ttl_status = None is a no op
+        return (
+            ttl_status.value
```

Comment:
Nit: `ArtifactTTLChange` has a different meaning than the values passed to GraphQL setter. Relying on some of the values being the same is error prone. Let's just do this explicitly:
```
if ttl_status is None:
    return None
if ttl_status == INHERITED:
    return -1
return ttl_duration_seconds or -2
```